easybuild/easyblocks/f/flexiblas.py

```diff
@@ -0,0 +1,137 @@
+##
+# Copyright 2021-2021 Ghent University
+#
+# This file is part of EasyBuild,
+# originally created by the HPC team of Ghent University (http://ugent.be/hpc/en),
+# with support of Ghent University (http://ugent.be/hpc),
+# the Flemish Supercomputer Centre (VSC) (https://www.vscentrum.be),
+# Flemish Research Foundation (FWO) (http://www.fwo.be/en)
+# and the Department of Economy, Science and Innovation (EWI) (http://www.ewi-vlaanderen.be/en).
+#
+# https://github.com/easybuilders/easybuild
+#
+# EasyBuild is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation v2.
+#
+# EasyBuild is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+##
+"""
+EasyBuild support for building and installing FlexiBLAS, implemented as an easyblock
+
+author: Kenneth Hoste (HPC-UGent)
+"""
+import os
+
+from easybuild.easyblocks.generic.cmakemake import CMakeMake
+from easybuild.framework.easyconfig import CUSTOM
+from easybuild.tools.build_log import EasyBuildError
+from easybuild.tools.systemtools import get_shared_lib_ext
+
+
+class EB_FlexiBLAS(CMakeMake):
+    """Support for building/installing FlexiBLAS."""
+
+    @staticmethod
+    def extra_options():
+        extra_vars = CMakeMake.extra_options()
+        extra_vars.update({
+            'blas_auto_detect': [False, "Let FlexiBLAS autodetect the BLAS libraries during configuration", CUSTOM],
+            'enable_lapack': [True, "Enable LAPACK support, also includes the wrappers around LAPACK", CUSTOM],
+            'flexiblas_default': [None, "Default BLAS lib to set at compile time. If not defined, " +
+                                  "the first BLAS lib in the list of dependencies is set as default", CUSTOM],
+        })
+        extra_vars['separate_build_dir'][0] = True
+        return extra_vars
+
+    def __init__(self, *args, **kwargs):
+        """Easyblock constructor."""
+        super(EB_FlexiBLAS, self).__init__(*args, **kwargs)
+
+        build_dep_names = set(dep['name'] for dep in self.cfg.dependencies(build_only=True))
+        dep_names = [dep['name'] for dep in self.cfg.dependencies()]
+        self.blas_libs = [x for x in dep_names if x not in build_dep_names]
+
+    def configure_step(self):
+        """Custom configuration for FlexiBLAS, based on which BLAS libraries are included as dependencies."""
+
+        configopts = {}
+
+        if self.cfg['blas_auto_detect'] is True:
+            configopts.update({'BLAS_AUTO_DETECT': 'ON'})
+        else:
+            configopts.update({'BLAS_AUTO_DETECT': 'OFF'})
```

Comment:
This is a bit inefficient. Better:

```suggestion
        if self.cfg['blas_auto_detect']:
            configopts['BLAS_AUTO_DETECT'] = 'ON'
        else:
            configopts['BLAS_AUTO_DETECT'] = 'OFF'
```

Or even: configopts['BLAS_AUTO_DETECT'] = ('OFF', 'ON')[bool(self.cfg['blas_auto_detect'])] then you could even use the initial dict creation:

```
        configopts = {
            'BLAS_AUTO_DETECT': ('OFF', 'ON')[bool(self.cfg['blas_auto_detect'])] 
            'LAPACK': ('OFF', 'ON')[bool(self.cfg['enable_lapack'])] 
            'FLEXIBLAS_DEFAULT': self.blas_libs[0],
        }
```

(and overwrite the last entry if required)

So generally: `is True` is not required and `dict.update` is inefficient for single entries because it creates a new dict and needs to merge those 2 then instead of a simple setting of a key-value

Reply:
Makes sense, I'll make that change.